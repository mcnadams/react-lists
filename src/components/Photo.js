import React from 'react';
import PropTypes from 'prop-types';

function Photo({ photoUrl }) {
  return <img src={photoUrl} style={{ 'width': '20vw' }}/>;
}

Photo.propTypes = {
  photoUrl: PropTypes.string.isRequired
};

export default Photo;
