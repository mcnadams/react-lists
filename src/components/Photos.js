import React from 'react';
import Photo from './Photo';
import PropTypes from 'prop-types';

function Photos({ photos }) {
  const photoList = photos.map(photo => {
    console.log(photo);
    return (
      <li key={photo} style={{ 'list-style': 'none', display: 'inline-block', margin: '10px' }}>
        <Photo photoUrl={photo} />
      </li>
    );
  });

  return (
    <ul>
      {photoList}
    </ul>
  );
}

Photos.propTypes = {
  photos: PropTypes.array.isRequired
};

export default Photos;
