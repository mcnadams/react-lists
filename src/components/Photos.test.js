import React from 'react';
import { shallow } from 'enzyme';
import Photo from './Photo';
import Photos from './Photos';

describe('Photos test', () => {
  it('make a list of photos', () => {
    const wrapper = shallow(<Photos photos={['./path', './path2']} />);
    expect(wrapper).toMatchSnapshot();
  });
});
