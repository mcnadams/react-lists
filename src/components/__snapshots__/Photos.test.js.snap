// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos test make a list of photos 1`] = `
<ul>
  <li
    key="./path"
    style={
      Object {
        "display": "inline-block",
        "list-style": "none",
        "margin": "10px",
      }
    }
  >
    <Photo
      photoUrl="./path"
    />
  </li>
  <li
    key="./path2"
    style={
      Object {
        "display": "inline-block",
        "list-style": "none",
        "margin": "10px",
      }
    }
  >
    <Photo
      photoUrl="./path2"
    />
  </li>
</ul>
`;
